source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -955,139 +1271,198 @@ namespace AdaptiveCards::Rendering::Uwp::ActionHelpers
+                                                                   renderArgs,
+                                                                   &actionControl));
+
+                    currentButtonIndex++;
                 }
-
-                if (columnDefinitions != nullptr)
+                else
                 {
-                    // If using the equal width columns, we'll add a column and assign the column
-                    ComPtr<IColumnDefinition> columnDefinition = XamlHelpers::CreateABIClass<IColumnDefinition>(
-                        HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_ColumnDefinition));
-                    RETURN_IF_FAILED(columnDefinition->put_Width({1.0, GridUnitType::GridUnitType_Star}));
-                    RETURN_IF_FAILED(columnDefinitions->Append(columnDefinition.Get()));
-
-                    ComPtr<IFrameworkElement> actionFrameworkElement;
-                    THROW_IF_FAILED(actionControl.As(&actionFrameworkElement));
-                    THROW_IF_FAILED(gridStatics->SetColumn(actionFrameworkElement.Get(), currentAction));
+                    pastLastPrimaryAction = true;
                 }
             }
-            else
+            else if (pastLastPrimaryAction && (mode == ABI::AdaptiveCards::ObjectModel::Uwp::ActionMode::Primary) && !overflowMaxActions)
             {
+                // If we have more primary actions than the max actions and we're not allowed to overflow them just set a warning and continue
                 renderContext->AddWarning(ABI::AdaptiveCards::ObjectModel::Uwp::WarningStatusCode::MaxActionsExceeded,
                                           HStringReference(L"Some actions were not rendered due to exceeding the maximum number of actions allowed")
                                               .Get());
+                return S_OK;
             }
-            currentAction++;
+            else
+            {
+                // If the action's mode is secondary or we're overflowing max actions, create a flyout item on the overflow menu
+                if (overflowButton == nullptr)
+                {
+                    // Create a button for the overflow menu if it doesn't exist yet
+                    RETURN_IF_FAILED(CreateFlyoutButton(renderContext, renderArgs, &overflowButton));
+                }
+
+                // Add a flyout item to the overflow menu
+                RETURN_IF_FAILED(AddOverflowFlyoutItem(action.Get(), overflowButton.Get(), renderContext, &actionControl));
+
+                // Show card actions in the overflow menu move to the action bar when selected, so we need to keep track
+                // of these so we can make non-visible buttons for them later. They need to go after all the actions we're creating as primary.
+                if (actionType == ActionType_ShowCard)
+                {
+                    showCardOverflowButtons.emplace_back(std::make_pair(action, actionControl));
```

Comment:
just a note -- IIRC you can replace the `std::make_pair(...)` here with `{action, actionControl}`. not sure if we have a standard convention on that or not though, so use what you prefer :)

Reply:
That doesn't immediately seem to compile, so leaving as is.